src/neml2/models/solid_mechanics/KocksMeckingIntercept.cxx

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "neml2/models/solid_mechanics/KocksMeckingIntercept.h"
+
+#include "neml2/misc/math.h"
+
+namespace neml2
+{
+register_NEML2_object(KocksMeckingIntercept);
+
+OptionSet
+KocksMeckingIntercept::expected_options()
+{
+  OptionSet options = NonlinearParameter<Scalar>::expected_options();
+
+  options.doc() = "The critical value of the normalized activation energy given by \\f$ g_0 "
+                  "\\frac{C-B}{A} \\f$";
+
+  options.set<CrossRef<Scalar>>("A");
+  options.set("A").doc() = "The Kocks-Mecking slope";
+
+  options.set<CrossRef<Scalar>>("B");
+  options.set("B").doc() = "The Kocks-Mecking intercept";
+
+  options.set<CrossRef<Scalar>>("C");
```

Comment:
```suggestion
  options.set_parameter<CrossRef<Scalar>>("C");
```
this is to set the metadata so that syntax documentation can mark A, B, C with that bold blue P emoji.